src/pybind/mgr/dashboard/frontend/src/app/shared/services/formatter.service.ts

```diff
@@ -28,7 +28,14 @@ export class FormatterService {
     }
     const unit = n < 1 ? 0 : Math.floor(Math.log(n) / Math.log(divisor));
     const truncatedFloat = this.truncate(n / Math.pow(divisor, unit), decimals);
-    return truncatedFloat === '' ? '-' : truncatedFloat + units[unit];
+    if (truncatedFloat === '') {
+      return '-';
+    }
+    let result = truncatedFloat;
+    if (units[unit] !== '') {
```

Comment:
This can only happen if the units array has an empty string in it.
It would be better to check the following:
```
units[units.length <= unit ? units.length -1 : unit]
```